### ISSUE TITLE
Documentation page for system properties

### ISSUE THREAD

**Issue by RaasAhsan** [typelevel/cats-effect#1127]
Since we're relying more and more on system properties to customize the behavior of `IO`, we should document them all in a dedicated doc page for discoverability.

**Comment by djspiewak** [comment]
Very much this

**Comment by i10416** [comment]
Any other properties?
- cats.effect.cancelation.check.threshold
- cats.effect.auto.yield.threshold.multiplier
- cats.effect.logNonDaemonThreadsOnExit
- cats.effect.logNonDaemonThreads.sleepIntervalMillis
- cats.effect.tracing.mode

series/3.x
- https://github.com/typelevel/cats-effect/blob/series/3.x/core/jvm/src/main/scala/cats/effect/unsafe/IORuntimeConfigCompanionPlatform.scala#L22
- https://github.com/typelevel/cats-effect/blob/series/3.x/core/jvm/src/main/scala/cats/effect/NonDaemonThreadLogger.scala#L28
- https://github.com/typelevel/cats-effect/blob/series/3.x/core/jvm/src/main/java/cats/effect/tracing/TracingConstants.java#L27

series/2.x
- ???

**Comment by vasilmkd** [comment]
`cats.effect.tracing.exceptions.enhanced`, `cats.effect.tracing.buffer.size` in https://github.com/typelevel/cats-effect/blob/series/3.x/core/jvm/src/main/scala/cats/effect/unsafe/IORuntimeConfigCompanionPlatform.scala

**Comment by i10416** [comment]
@vasilmkd Thank you.
I made PR to get todo list up-to-date https://github.com/typelevel/cats-effect/pull/2398 and going to send another PR to add documentation. I'm considering putting a doc page for system property in "Scaling and Tuning" section. What do you think?

**Comment by vasilmkd** [comment]
Go for it @i10416.

**Comment by armanbilge** [comment]
We should also list the equivalent environmental variable names for the JS runtimes as well on this page, but only after they are released.
